sql/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java

```diff
@@ -1691,4 +1691,20 @@ public void test_values_as_top_level_relation() {
                "3| c\n")
         );
     }
+
+
+    @Test
+    public void test_select_with_explicit_high_limit_and_query_that_does_not_match_anything_to_trigger_batch_size_optimization() {
+        execute("create table tbl (x int) clustered into 1 shards");
+        Object[][] values = new Object[2000][];
+        for (int i = 0; i < values.length; i++) {
+            values[i] = new Object[] { i };
+        }
+        execute("insert into tbl (x) values (?)", values);
```

Comment:
This is a bit heavy, but the shard needs to have > 1000 documents to take the branch which contained the bug.